curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-destructuring-assignment-with-the-rest-parameter-to-reassign-array-elements.md

```diff
@@ -33,33 +33,27 @@ Use a destructuring assignment with the rest syntax to emulate the behavior of `
 `removeFirstTwo([1, 2, 3, 4, 5])` should be `[3, 4, 5]`
 
 ```js
-const testArr_ = [1, 2, 3, 4, 5];
-const testArrWORemoved_ = removeFirstTwo(testArr_);
-assert(testArrWORemoved_.every((e, i) => e === i + 3) && testArrWORemoved_.length === 3);
+assert.deepEqual(removeFirstTwo([1, 2, 3, 4 5]), [3, 4, 5]);
 ```
 
 `removeFirstTwo()` should not modify `list`
 
 ```js
-const testArr_ = [1, 2, 3, 4, 5];
-const testArrWORemoved_ = removeFirstTwo(testArr_);
-assert(testArr_.every((e, i) => e === i + 1) && testArr_.length === 5);
+const _testArr = [1, 2, 3, 4, 5];
+removeFirstTwo(_testArr);
+assert.deepEqual(_testArr, [1, 2, 3, 4, 5])
 ```
 
 `Array.slice()` should not be used.
 
 ```js
-assert(!code.match(/slice/g));
+assert(!code.match(/\.\s*slice\s*\(/));
 ```
 
-Destructuring on `list` should be used.
+You should use the rest syntax.
 
 ```js
-assert(
-  __helpers
-    .removeWhiteSpace(code)
-    .match(/\[(([_$a-z]\w*)?,){1,}\.\.\.shorterList\]=list/i)
-);
+assert.match(code, /\.\s*\.\s*\./);
```

Comment:
Ideally, we would be using an AST looking for an ArrayPattern: https://astexplorer.net/#/gist/8074738b8ae4315e9018854307a82946/920c3d92d966405a231b4562c8264a2b330266ee

short of that, we might need to write a Regex looking for `..` followed by all valid variable identifiers.

Reply:
That seems overkill. Is there a case where a camper could use a spread operator without spreading a variable that wouldn't throw an error?

Reply:
Fair enough.

As far as I can tell, there is not allowed to be whitespace between the `.` because they form an operator: https://tc39.es/ecma262/multipage/ecmascript-language-expressions.html#prod-AssignmentRestElement

```suggestion
assert.match(code, /\.\.\./);
```

Reply:
I could have sworn there was a challenge where we did this. 🤦🏻‍♀️